docs/DESKTOP_ENVIRONMENTS.md

```diff
@@ -0,0 +1,118 @@
+applications. This could e.g. mean reserving memory to session processes,
+preferentially killing background tasks in out-of-memory situations
+or assinging different memory/CPU/IO priorities to ensure that the session
+runs smoothly under load.
+
+TODO: Will there be a default to place units into e.g. `apps.slice` by default
+rather than the root slice?
+
+## XDG standardization for applications
+
+To ensure cross-desktop compatibility and encourage sharing of good practices,
+desktop environments should adhere to the following conventions:
+
+ * Application units should follow the scheme `apps-launcher-ApplicationID-PID.service`,
+   e.g. `apps-gnome-org.gnome.Evince-12345.service`,
+   `apps-flatpak-org.telegram.desktop-12345.service` or `apps-KDE-org.kde.okular-12345.service`.
+ * Use of `.scope` units is discouraged.
+ * Applications must be placed into the appropriate slice. It is suggested to use
+   the same naming scheme, and use the `apps-launcher-ApplicationID.slice`
+   (the PID may be included optionally, simplifying the unit creation as it
+   can be done by setting `Slice=%N.slice` in a drop-in configuration).
+ * If no application ID is available, the launcher should generate a reasonable
+   name when possible (e.g. using `basename(argv[0])`).
+
+TODO: Define the name of slices that should be used. This could be `apps-launcher-ApplicationID.slice`.
+
+This has the following advantages:
+ * Using the `apps-launcher-` prefix means that the unit defaults can be adjusted
+   using desktop environment specific drop-in files.
+ * The application ID can be retrieved by stripping the prefix and postfix.
+   This in turn should map to the corresponding `.desktop` file when available
+
+## XDG autostart integration
+
+To allow XDG autostart integration, systemd will ship a cross-desktop generator
+to create appropriate units for the autostart directory. Desktop Environments
+will be able to make use of this simply by starting the appropriate XDG related
+targets. The names and ordering rules for these targets are to be defined.
+
+This generator will likely only support a subset of features. In particular,
+the GNOME specific feature to bind a service to a settings variable will not
+be supported.
+
+## Application managed cgroups
+
+Applications like browsers should place major components that are run as
+separate processes into their own cgroup. This can be useful to the
+applications themselves to apply further resource restrictions (e.g. disallow
+network access or memory limits). Furthermore, it exposes the structure of
+the application to the system, allowing it to e.g. kill entire cgroups in
+response to OOM situations without affecting the application as a whole.
+
+It is to be defined how these applications should create and manage these
+cgroups. Possible approaches could be:
+
+ * Directly use systemd and explicitly place items into a common slice. This
+   has the disadvantage of requiring privileged access to systemd.
```

Comment:
Hmm, any user process can talk to systemd --user. No extra privileges are generally necessary.

Reply:
New sentences on new lines...

Reply:
From the point of view of systemd, yes. But I think we want to give Flatpak'ed applications access to such facilities through an XDG portal. And then it makes sense to encourage the portal use even if the direct use of systemd is legitimate.